cmd/minishift/cmd/addon/addon_enable_test.go

```diff
@@ -19,20 +19,21 @@ package addon
 import (
 	"bytes"
 	"fmt"
-	"github.com/minishift/minishift/pkg/testing/cli"
-	"github.com/minishift/minishift/pkg/util/os/atexit"
 	"io"
 	"os"
 	"path/filepath"
 	"testing"
+
+	"github.com/minishift/minishift/pkg/testing/cli"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
 )
 
 func Test_addon_name_must_be_specified_for_enable_command(t *testing.T) {
 	tmpMinishiftHomeDir := cli.SetupTmpMinishiftHome(t)
-	origStdout, stdOutWriter, stdOutReader := cli.CaptureStdOut(t)
-	defer cli.TearDown(tmpMinishiftHomeDir, origStdout)
+	origStdout, origStderr, streamWriter, streamReader := cli.CaptureStreamOut(t, 1)
```

Comment:
this feels just wrong. First one has so many return values and second you need to pass the exit code here which somehow decides which stream to capture. This is confusing at best. I think I have an idea to fix that. We need to encapsulate this into a struct. This will also make the teardown easier. 

I can take it from here if you like unless you want to give it a go.